Hexalem/src/net/Substrate.Hexalem.NET/Substrate.Hexalem.NET/HexaBoard.cs

```diff
@@ -331,8 +216,189 @@ internal bool Place((int, int) coords, HexaTile chooseTile)
+
+                            tile.SetFormationFlag1 = (delta.HasValue && delta.Value.Item1 == TileType.Tree && delta.Value.Item2 == TileType.Tree);
+
+                            var reverseDelta = GetReverseDeltaPosition(treeNeighbours);
+
+                            tile.SetFormationFlag2 = (reverseDelta.HasValue && reverseDelta.Value.Item1 == TileType.Tree && reverseDelta.Value.Item2 == TileType.Tree);
+
+                            Value[index] = tile;
+
+                            break;
+
+                        case TileType.Mountain:
+                            var mountainNeighbours = GetNeighbors(c.Value);
+
+                            var ypsilon = GetYpsilonPosition(mountainNeighbours);
+
+                            tile.SetFormationFlag1 = (ypsilon.HasValue && ypsilon.Value.Item1 == TileType.Mountain && ypsilon.Value.Item2 == TileType.Mountain && ypsilon.Value.Item3 == TileType.Mountain);
+
+                            var reverseYpsilon = GetReverseYpsilonPosition(mountainNeighbours);
+
+                            tile.SetFormationFlag2 = (reverseYpsilon.HasValue && reverseYpsilon.Value.Item1 == TileType.Mountain && reverseYpsilon.Value.Item2 == TileType.Mountain && reverseYpsilon.Value.Item3 == TileType.Mountain);
+
+                            Value[index] = tile;
+
+                            break;
+
+                        case TileType.Water:
+                            var waterNeighbours = GetNeighbors(c.Value);
+
+                            var lineRight = GetLineRightPosition(waterNeighbours);
+
+                            tile.SetFormationFlag1 = (lineRight.HasValue && lineRight.Value.Item1 == TileType.Water && lineRight.Value.Item2 == TileType.Water);
+
+                            var lineLeft = GetLineLeftPosition(waterNeighbours);
+
+                            tile.SetFormationFlag2 = (lineLeft.HasValue && lineLeft.Value.Item1 == TileType.Water && lineLeft.Value.Item2 == TileType.Water);
+
+                            Value[index] = tile;
+
+                            break;
+
+                        case TileType.Grass:
+                            var grassNeighbours = GetNeighbors(c.Value);
+
+                            var grassDelta = GetDeltaPosition(grassNeighbours);
+
+                            tile.SetFormationFlag1 = (grassDelta.HasValue && grassDelta.Value.Item1 == TileType.Water && grassDelta.Value.Item2 == TileType.Grass);
+
+                            var grassReverseDelta = GetReverseDeltaPosition(grassNeighbours);
+
+                            tile.SetFormationFlag2 = (grassReverseDelta.HasValue && grassReverseDelta.Value.Item1 == TileType.Grass && grassReverseDelta.Value.Item2 == TileType.Water);
+
+                            Value[index] = tile;
+
+                            break;
+                    }
+
+                }
+            }
+        }
+
+        internal (TileType, TileType)? GetDeltaPosition(List<(int, int)?> neighbours)
```

Comment:
This doesn't account for all variations with 2 grass and 1 water in a delta formation.